### PR TITLE
PMM-6716 Add ShowXtraDBCluster RPC.

### PR DESCRIPTION
**Pull request by askomorokhov**
https://jira.percona.com/browse/PMM-6716

FB: https://github.com/Percona-Lab/pmm-submodules/pull/1177

- https://github.com/percona-platform/dbaas-controller/pull/65
- https://github.com/Percona-Lab/pmm-api-tests/pull/132
- https://github.com/percona/pmm-managed/pull/490
- https://github.com/percona/pmm/pull/594

### CODE REVIEW

**Review by percona-csalguero — APPROVED** [review]

**Review by JiriCtvrtka — APPROVED** [review]

**Review by BupycHuk — APPROVED** [review]

File: api/managementpb/dbaas/xtradb_cluster.proto [hunk 1]
```diff
@@ -55,21 +55,13 @@ enum XtraDBBackupState {
 message XtraDBClusterParams {
   // PXC container parameters.
   message PXC {
-    // Requested compute resources.
-    ComputeResources compute_resources = 1 [
-      (validator.field) = {
-        msg_exists: true
-      }
-    ];
+    // Requested compute resources. Do not pass for minikube.
```
Comment (AlekSi) [inline api/managementpb/dbaas/xtradb_cluster.proto:58]: Well, that's not accurate – minikube _can_ enforce limits, and one _can_ pass them.

File: api/managementpb/dbaas/xtradb_cluster.proto [hunk 2]
```diff
@@ -129,6 +121,43 @@ message CreateXtraDBClusterRequest {
       string_not_empty: true
     }
   ];
+}
+
+message ShowXtraDBClusterResponse {
+  // Cluster name.
+  string name = 1;
```
Comment (AlekSi) [inline api/managementpb/dbaas/xtradb_cluster.proto:128]: Why we return cluster name in "show cluster by name" request?

File: api/managementpb/dbaas/xtradb_cluster.proto [hunk 3]
```diff
@@ -129,6 +121,43 @@ message CreateXtraDBClusterRequest {
...
+  // Cluster parameters.
+  XtraDBClusterParams params = 4;
+  // ProxySQL username.
+  string username = 5;
+  // ProxySQL password.
+  string password = 6;
+  // ProxySQL host.
+  string host = 7;
+  // ProxySQL port.
+  int32 port = 8;
```
Comment (AlekSi) [inline api/managementpb/dbaas/xtradb_cluster.proto:142]: Those 4 parameters should be a separate message type as we will likely return several sets of credentials

File: api/managementpb/dbaas/xtradb_cluster.proto [hunk 4]
```diff
@@ -188,6 +217,13 @@ service XtraDBCluster {
       body: "*"
     };
   }
+  // ShowXtraDBCluster returns a XtraDB cluster by name.
+  rpc ShowXtraDBCluster(ShowXtraDBClusterRequest) returns (ShowXtraDBClusterResponse) {
+    option (google.api.http) = {
+      post: "/v1/management/DBaaS/XtraDBClusters/Show"
```
Comment (AlekSi) [inline api/managementpb/dbaas/xtradb_cluster.proto:223]: Why it is `/Show` and not `/Get` as in other APIs?

**Review by AlekSi — CHANGES_REQUESTED** [review]

File: api/managementpb/dbaas/xtradb_cluster.proto [hunk 5]
```diff
@@ -129,6 +121,43 @@ message CreateXtraDBClusterRequest {
...
+      string_not_empty: true
+    }
+  ];
+  // XtraDB cluster name.
+  // a DNS-1035 label must consist of lower case alphanumeric characters or '-',
+  // start with an alphabetic character, and end with an alphanumeric character
+  // (e.g. 'my-name',  or 'abc-123', regex used for validation is '[a-z]([-a-z0-9]*[a-z0-9])?')
+  string name = 2 [
+    (validator.field) = {
+      regex: "^[a-z]([-a-z0-9]*[a-z0-9])?$"
```
Comment (AlekSi) [inline api/managementpb/dbaas/xtradb_cluster.proto:158]: That validation really should be on dbaas-controller side

File: api/managementpb/dbaas/xtradb_cluster.proto (hunk 3 above)
Comment (BupycHuk) [reply api/managementpb/dbaas/xtradb_cluster.proto:142]: what do you mean by several sets of credentials?

Comment (AlekSi) [reply api/managementpb/dbaas/xtradb_cluster.proto:142]: Several username/password pairs for different access rights permissions. Several host/port pairs for different access methods (direct connections vs going through the proxy). Etc.

We shouldn't make this field a repeated field now, but let's extract a separate message type

**Review by AlekSi — APPROVED** [review]

**Pull request merged**